apps/04TowersOfHanoi.js

```diff
@@ -19,23 +19,33 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  var storage = stacks[startStack].pop();
+  stacks[endStack].push(storage);
 }
 
-function isLegal() {
-  // Your code here
-
+function isLegal(startStack, endStack) {
+  if  (stacks[startStack[startStack.length - 1]] < stacks[endStack[endStack.length - 1]] || stacks[endStack].length === 0) {
+    movePiece(startStack, endStack);
```

Comment:
This does work but see if you can find a way to get it to work without having movePiece() inside of isLegal